interfaces/builtin/kubernetes_support.go

```diff
@@ -79,13 +79,23 @@ profile systemd_run (attach_disconnected,mediate_deleted) {
   # setsockopt()
   capability net_admin,
 
-  # ptrace 'trace' is coarse and not required for using the systemd private
-  # socket, and while the child profile omits 'capability sys_ptrace', skip
-  # for now since it isn't strictly required.
-  ptrace read peer=unconfined,
-  deny ptrace trace peer=unconfined,
+  # kubelet calls 'systemd-run --scope true' to determine if systemd is
+  # available and usable for calling certain mount commands under transient
+  # units as part of its lifecycle management. This requires ptrace 'read' on
+  # unconfined since systemd-run will call its detect_container() which will
+  # try to read /proc/1/environ. This is mediated via PTRACE_MODE_READ when
+  # run within kubelet's namespace.
+  ptrace (read) peer=unconfined,
   /run/systemd/private rw,
 
+  # Ubuntu's ptrace patchset before (at least) 20.04 did not correctly evaluate
+  # PTRACE_MODE_READ and policy required 'trace' instead of 'read'.
+  # (LP: #1890848). This child profile doesn't have 'capability sys_ptrace', so
+  # continue to allow this historic 'trace' rule on unconfined (which systemd
+  # runs as) since systemd-run won't be able to ptrace this snap's processes.
+  # This can be dropped once LP: #1890848 is fixed.
+  ptrace (trace) peer=unconfined,
```

Comment:
Your understanding is correct. There shouldn't be an issue because someone would need to be to have forked our xenial or bionic kernel and not update it. Flavors will take our kernel and derivatives typically either take our kernel wholesale (with -security and -updates flowing in) or use their own kernel without the AppArmor SAUCE (I recently did a survey for what distros support strict mode snaps and this is my conclusion). Furthermore, a distro that happened to take our SAUCE patches is likely to already be out of support.